### PR TITLE
Updated to Docker README

### PR DESCRIPTION
**Pull request by ashgillman**
- Moved Introduction to straight after TL;DR
- Outlined in a few relevant places how to re-build. (what about re-pull?)
- I found I needed to build using the convenience function, not just `docker-compose`, on Linux.
- Other minor edits

Please check for accuracy! I've tried hard to fully understand how it's all working here - but I didn't write these configurations and convenience wrappers.

### CODE REVIEW

**Comment by ashgillman** [comment]
Fixes #584 ?

File: docker/README.md [hunk 1]
```diff
@@ -150,34 +154,37 @@ files and notebooks in `/devel` will be persistent between sessions and
...
 
-We can now use this interactively.
+Here, `--no-start` delays actually starting the container.
+We can now use this interactively, by starting the containder with flags `-ai`.
 ```bash
 SIRF-SuperBuild/docker$ docker start -ai sirf
-(py2) sirf:~$ gadgetron >> /dev/null &
-(py2) sirf:~$ python SIRF-SuperBuild/SIRF/examples/Python/MR/fully_sampled_recon.py
+(py2) sirf:~$ gadgetron >> /dev/null &  # we are free to use tools like Gadgetron
+(py2) sirf:~$ python SIRF-SuperBuild/SIRF/examples/Python/MR/fully_sampled_recon.py  # or SIRF
```
Comment (KrisThielemans) [inline docker/README.md:182]: I don't think I would say this. They need to launch `gadgetron` before doing (MR) SIRF. These mods don't make that clear.

File: docker/README.md [hunk 2]
```diff
@@ -150,34 +154,37 @@ files and notebooks in `/devel` will be persistent between sessions and
...
 (py2) sirf:~$ exit
 ```
 
 The first line starts the `sirf` docker container.
-The second line starts `gadgetron` within the container as a background process.
+The second line starts `gadgetron` within the container as a background process (optional, but needed for using Gadgetron of course).
```
Comment (KrisThielemans) [inline docker/README.md:187]:
```suggestion
The second line starts `gadgetron` within the container as a background process (optional, but needed for using Gadgetron, i.e. most SIRF MR functionality, of course).
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
thanks! This helps. A few tiny comments

File: docker/README.md (hunk 1 above)
Comment (KrisThielemans) [reply docker/README.md:182]:
```suggestion
(py2) sirf:~$ gadgetron >> /dev/null &  # launch Gadgetron as a "background" process
(py2) sirf:~$ python SIRF-SuperBuild/SIRF/examples/Python/MR/fully_sampled_recon.py  # run a SIRF demo
```

Comment (KrisThielemans) [reply docker/README.md:182]: actually, that line didn't work!
```suggestion
(py2) sirf:~$ gadgetron >> /dev/null &  # launch Gadgetron as a "background" process
(py2) sirf:~$ python $SIRF_PATH/examples/Python/MR/fully_sampled_recon.py  # run a SIRF demo
```

**Pull request merged**